[FSDP][Docs] Document get_state_dict_type

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102890
* #102692
* #102680
* #102672
* __->__ #102658
* #102657
* #102639
* #102637
* #102508

Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

rohan-varma committed:
[FSDP][Docs] Document get_state_dict_type

Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

[ghstack-poisoned]

fegin reviewed (approved):
Nice!

rohan-varma committed:
Update on "[FSDP][Docs] Document get_state_dict_type"


Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP][Docs] Document get_state_dict_type"


Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
+        Get the state_dict_type and the corresponding configurations
+        for the FSDP modules rooted at ``module``. The target module
+        does not have to be an FSDP module. Note that this method will
+        throw if the ``StateDictSettings`` for differenet FSDP submodules
```
```suggestion
        throw if the ``StateDictSettings`` for different FSDP submodules
```

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
+        differ.
+
+        Returns:
+            A ``StateDictSettings`` containing the state_dict_type and
+            state_dict / optim_state_dict configs that are currently set.
```
nit: Same with "state_dict_type", "state_dict", and "optim_state_dict" here.

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
     @staticmethod
     def get_state_dict_type(module: nn.Module) -> StateDictSettings:
+        """
+        Get the state_dict_type and the corresponding configurations
```
nit: We should use double backtick for "state_dict_type" or use "state dict type" without underscores.

rohan-varma replied:
Will just leave them as non-backticked

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
+        """
+        Get the state_dict_type and the corresponding configurations
+        for the FSDP modules rooted at ``module``. The target module
+        does not have to be an FSDP module. Note that this method will
```
nit: Should we just document via `Raises:` in the docstring the condition for raising an error?

awgu reviewed (approved):
Stamp to unblock

rohan-varma committed:
Update on "[FSDP][Docs] Document get_state_dict_type"


Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP][Docs] Document get_state_dict_type"


Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

[ghstack-poisoned]
